### ISSUE TITLE
http://itsat.info: expired but still censored

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe#2308]
**TL;DR** The `http://itsat.info` domain does not exist anymore but there's still residual DNS censorship in Italy. I like to think at residual DNS censorship as radiation. Anyway, we probably want to remove it from the Italian's test list.

The `http://itsat.info` domain has expired (see [#1](https://explorer.ooni.org/measurement/20220914T071816Z_urlgetter_IT_30722_n1_crHiO4euUC7Ay3o7?input=dnslookup%3A%2F%2Fitsat.info), [#2](https://explorer.ooni.org/measurement/20220914T071902Z_urlgetter_IT_30722_n1_trZuJxVllIP4Mbok?input=dnslookup%3A%2F%2Fitsat.info), [#3](https://explorer.ooni.org/measurement/20220914T071911Z_urlgetter_IT_30722_n1_yIAasl3xulCG8NPh?input=dnslookup%3A%2F%2Fitsat.info)). See also [this whois result](https://who.is/whois/itsat.info):

<img width="818" alt="Screenshot 2022-09-14 at 09 22 53" src="https://user-images.githubusercontent.com/337298/190088096-aecaf1bf-79d4-4970-8a0c-4de563310200.png">

However, there is still residual censorship for this domain in a bunch of Italian ISPs. So, let us take a look at how this domain [appears in the MAT]():

<img width="1262" alt="Screenshot 2022-09-14 at 09 24 07" src="https://user-images.githubusercontent.com/337298/190088393-64c1b66c-7579-46bb-877b-c62e393e24fa.png">

* with v0.4 it's reported as "up" in [AS3269](https://explorer.ooni.org/measurement/20220914T031852Z_webconnectivity_IT_3269_n1_cJGaSVusHaEZ8o5S?input=http%3A%2F%2Fitsat.info%2F) because the system resolver is using Google's DNS. This result is _wrong_ but we know that v0.4 does not detect the case where there is consistency of NXDOMAIN.

* with v0.4 it's reported as "anomaly" in [AS12874](https://explorer.ooni.org/measurement/20220913T232651Z_webconnectivity_IT_12874_n1_MYWKuNxotSvp2imd?input=http%3A%2F%2Fitsat.info%2F) when using AS12874's own resolver (which resolves the address to 127.0.0.1)

* with v0.45 it's reported as "up" in [AS30722](https://explorer.ooni.org/measurement/20220913T200719Z_webconnectivity_IT_30722_n1_5iHPxGy6tVskeqmQ?input=http%3A%2F%2Fitsat.info%2F) when again we're using Google's DNS

Now, what should be the correct analysis for this measurement? So, it turns out what we should actually say is that the domain does not exist anymore, but there is residual censorship. We need to detect this case in v0.5. 

Once we have fixed v0.5, **we should probably zap this domain from the test list**.

With an upcoming fix applied, v0.5 emits this:

```
[      0.923629] <info> [2/2] running with input: http://itsat.info
[      0.923851] <info> DNS whoami for 8.8.4.4:53/udp resolver: [{Address:91.80.36.88}]
[      0.923916] <info> DNS whoami for system resolver: [{Address:91.80.36.88}]
[      0.942759] <info> [#1] lookup itsat.info using 8.8.4.4:53... ok
[      0.943499] <info> [#2] lookup itsat.info using system... ok
[      0.943591] <info> using resolved addrs: [{Addr:83.103.74.20 Flags:3}]
[      0.943617] <info> prioritySelector: create with [{Addr:83.103.74.20 Flags:3}]
[      0.976184] <info> control for http://itsat.info using https://1.th.ooni.org... ok
[      0.976226] <info> additional addrs discovered by the TH: []
[      1.443987] <info> [#3] GET http://itsat.info using 83.103.74.20:80... in progress
[      1.444081] <info> [#4] GET https://itsat.info using 83.103.74.20:443... in progress
[     10.945081] <info> [#4] GET https://itsat.info using 83.103.74.20:443... generic_timeout_error
[     10.945081] <info> [#3] GET http://itsat.info using 83.103.74.20:80... generic_timeout_error
[     10.945183] <info> DNSConsistency: consistent
[     10.945436] <info> DNS censorship: local DNS success with remote NXDOMAIN
[     10.945447] <warn> RESIDUAL_DNS_BLOCKING: flags=0, accessible=false, blocking=dns
```

and [we obtain this measurement](https://explorer.ooni.org/measurement/20220914T074614Z_webconnectivity_IT_30722_n1_Afzy8plDB25oTbxJ?input=http%3A%2F%2Fitsat.info) that correctly flags the issue as being the DNS.
